src/mkfs.fat.c

```diff
@@ -1679,15 +1691,24 @@ int main(int argc, char **argv)
 	    sector_size = devinfo.sector_size;
 	    sector_size_set = 1;
 	}
+
+        if (devinfo.size <= part_offset)
+          die("The device %s size %llu is less then the offset %lld",
+              device_name, devinfo.size, (long long) part_offset);
     }
 
     if (sector_size > 4096)
 	fprintf(stderr,
 		"Warning: sector size %d > 4096 is non-standard, filesystem may not be usable\n",
 		sector_size);
 
-    cblocks = devinfo.size / BLOCK_SIZE;
-    orphaned_sectors = (devinfo.size % BLOCK_SIZE) / sector_size;
+    if (part_offset % sector_size)
```

Comment:
This looks suspicious. If offset is now in number of sectors then this does not make much sense...